src/plugins/pluginHooks.ts

```diff
@@ -18,7 +18,12 @@ export const usePluginSlotList = (name: string, defaults: any[] = []) => {
   // by default, all elements are visible,
   // unless visibility is configured explicitly in the metadata
   const elements = useMemo(
-    () => orderArray([...defaults, ...scope.getSlot(name)].filter(x => !x.metadata?.visible || x.metadata.visible())),
+    () =>
+      orderArray(
+        [...defaults, ...scope.getSlot(name)].filter(
+          x => !x.metadata?.visible || x.metadata.visible(scope.getState(name))
```

Comment:
Wasn't `metadata.visible` without params?

Reply:
needed state within visibility. but we can get it directly in the plugin definition, since we have access to `scope.getState`. no need for param

Reply:
Exactly :) I double-checked, and `visible` doesn't contain `params`, so this change is not necessary.